lightning/src/chain/keysinterface.rs

```diff
@@ -1074,15 +1080,26 @@ impl KeysManager {
 					}
 					spend_tx.input[input_idx].witness = keys_cache.as_ref().unwrap().0.sign_dynamic_p2wsh_input(&spend_tx, input_idx, &descriptor, &secp_ctx)?;
 				},
-				SpendableOutputDescriptor::StaticOutput { ref output, .. } => {
+				SpendableOutputDescriptor::StaticOutput { ref output, channel_keys_id,.. } => {
+					
+					let destination_script = channel_keys_id.map_or_else(|| self.destination_script.clone(),|s| self.get_destination_script(s));
 					let derivation_idx = if output.script_pubkey == self.destination_script {
 						1
+					} else if output.script_pubkey == destination_script && channel_keys_id.is_some() {
+						byte_utils::slice_to_be64(&channel_keys_id.unwrap()[0..8]) as u32
 					} else {
 						2
 					};
+
+					let master_derivation_idx = if self.new_style_derivation{
```

Comment:
We shouldn't be looking at `new_style_derivation` at all in this method, I think - we can totally have something that had the keys picked in the current version of LDK, but which needs to be spent in a new version with this code. In that case, we have to decide which derivation to use purely based on the `output.script_pub_key` field, not the setting here.